src/RateLimiter/CounterStatistics.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Web\RateLimiter;
+
+/**
+ * Rate limiter counter statistics
+ */
+final class CounterStatistics
+{
+    private int $limit;
+    private int $remaining;
+    private int $reset;
+
+    /**
+     * @param int $limit the maximum number of requests allowed with a time period
+     * @param int $remaining the number of remaining requests in the current time period
+     * @param int $reset milliseconds to wait until the rate limit resets
+     */
+    public function __construct(int $limit, int $remaining, int $reset)
+    {
+        $this->limit = $limit;
+        $this->remaining = $remaining;
+        $this->reset = $reset;
+    }
+
+    /**
+     * @return int the maximum number of requests allowed with a time period
+     */
+    public function getLimit(): int
+    {
+        return $this->limit;
+    }
+
+    /**
+     * @return int the number of remaining requests in the current time period
+     */
+    public function getRemaining(): int
+    {
+        return $this->remaining;
+    }
+
+    /**
+     * @return int milliseconds to wait until the rate limit resets
+     */
+    public function getReset(): int
```

Comment:
I suggests 
- `@return float seconds ...` `float` with second unit, OR
- use absolute timestamp: It is more accurate because the sending of the response causes a delay. In this case too, float is a better choice if seconds are not enough.

Reply:
Seems common practice is to use seconds: https://stackoverflow.com/questions/16022624/examples-of-http-api-rate-limiting-http-response-headers. Not sure if millisecond precision is needed.

Reply:
Absolute timestamp isn't common practice.

Reply:
@romkatsu what do you think?

Reply:
This class is not dedicated to HTTP headers. When you actually set the headers, you can calculate the absolute, relative, and any other mode.+

Reply:
```php
    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
    {
        $this->counter->setId($this->generateId($request));
        $result = $this->counter->incrementAndGetResult();  // <--- result.reset = 3 seconds
        if ($result->isLimitReached()) {
            $response = $this->createErrorResponse();
        } else {
            $response = $handler->handle($request);      // <-------- it runs in 10 seconds
        }
        return $this->addHeaders($response, $result);     // <---- The header will be 3 seconds, but the window expiration time is already 0 seconds.
    }
```

Reply:
@kamarton makes sense.

Reply:
fixed. now returns the time